### PR TITLE
Feat/404

### PR DESCRIPTION
**Pull request by zacwalters4**
1. What's this PR do?
- [ ] Adds 404 error handling for dead end urls
- [ ] Changes application title to Listen Up!
- [ ] Adds loading message and minor error handling to the game component
2. Where should the reviewer start?
- [ ] The new ErrorDisplay component
- [ ] The render and useEffect portions of the Game component
- [ ] The useEffect portion of the Difficulty component 
3. How should this be manually tested?
- [ ] Manually type in fake categories and difficulties to ensure that the user is routed to the 404 page
- [ ] Force an error in the Game component fetch call to ensure the user can see a message saying something went wrong
- [ ] Check if the loading screen pops up when the Game's fetch call is in progress 
4. Any background context you want to provide?
- I snuck a few minor changes into this PR because I don't think they warrant entire PRs
- The error handling on the game component definitely needs to be more fleshed out but I am not sure the best way to go about it
5. What are the relevant tickets?
https://github.com/orgs/Listen-Up-2210/projects/1/views/1?pane=issue&itemId=24500856
6. Screenshots (if appropriate)
image
7. Questions:

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -24,7 +24,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React App</title>
+    <title>Listen Up!</title>
   </head>
```
Comment (MRowan121) [inline public/index.html:28]: Nice catch on changing title to Listen Up

File: src/Components/Difficulty/Difficulty.jsx [hunk 2]
```diff
@@ -1,14 +1,24 @@
 import React from "react";
+import { useEffect } from "react"
 import "./Difficulty.css"
-import { Link, useLocation } from 'react-router-dom'
+import { Link, useLocation, useNavigate } from 'react-router-dom'
 
 function Difficulty() {
 
+    const validCategories = ['/animals', 'instruments', '/machines', '/misc']
```
Comment (MRowan121) [inline src/Components/Difficulty/Difficulty.jsx:8]: Update to `/instruments` in the validCategories array?

File: public/index.html
Comment (awirth224) [inline public/index.html:1]: Thanks for remembering to change the title tag!

File: src/Components/ErrorDisplay/ErrorDisplay.css [hunk 3]
```diff
@@ -0,0 +1,58 @@
+.error-display
+{
+	align-items: center;
```
Comment (nkinsaul) [inline src/Components/ErrorDisplay/ErrorDisplay.css:3]: Can we take out the extra spaces in this file and make sure everything is aligned the same?

File: src/Components/ErrorDisplay/ErrorDisplay.jsx
Comment (awirth224) [inline src/Components/ErrorDisplay/ErrorDisplay.jsx:1]: Lets add "=" after the ">" on lines 10 & 13. otherwise, looks great!

**Review by awirth224 — CHANGES_REQUESTED** [review]

File: src/Components/Difficulty/Difficulty.jsx (hunk 2 above)
Comment (BobAiken) [inline src/Components/Difficulty/Difficulty.jsx:8]: Does instruments needs a "/" in front of it?

File: src/Components/ErrorDisplay/ErrorDisplay.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    const error = props.errorCode
+    let errorMessage = ''
+
+    if(Number(error) > 500) {
+        errorMessage = 'Please try again later.'
+    }
+    else if(Number(error) > 400) {
+        errorMessage = 'The page you were looking for was not found.'
+    }
+    else{
```
Comment (nkinsaul) [inline src/Components/ErrorDisplay/ErrorDisplay.jsx:16]: Super nit-picky but just add a space after the else here for the sake of consistency :)

**Review by nkinsaul — CHANGES_REQUESTED** [review]
Thanks Zac!

**Pull request merged**